Bridged network setup fails on arping check step on Alpine Linux

While trying to setup bridged network on Alpine Linux v3.7, the initialization procedure [`setup_network`](https://github.com/tailhook/lithos/blob/master/src/bin/lithos_knot/setup_network.rs#L199) fails while checking interface avalibility with `arping` on the following error:
#

 `Fatal error running "container/container.0": arping failed: exited with code 1`

Full log of the container attache below:

``` text
[2018-06-19T11:46:53Z][DEBUG]src/mount.rs:148: Making private "/"
[2018-06-19T11:46:53Z][DEBUG]src/mount.rs:132: Remount readonly: "/run/lithos/mnt"
[2018-06-19T11:46:53Z][INFO] [container/container.0] Starting container
[2018-06-19T11:46:53Z][DEBUG]src/bin/lithos_knot/setup_network.rs:106: Running "ip" "link" "add" "li_ca02f9_0001" "type" "veth" "peer" "name" "li-ca02f9-0001"
[2018-06-19T11:46:53Z][DEBUG]src/bin/lithos_knot/setup_network.rs:119: Running "ip" "link" "set" "dev" "li_ca02f9_0001" "netns" "/proc/28231/fd/6"
[2018-06-19T11:46:53Z][DEBUG]src/bin/lithos_knot/setup_network.rs:131: Running "brctl" "addif" "br0" "li_ca02f9_0001"
[2018-06-19T11:46:53Z][DEBUG]src/bin/lithos_knot/setup_network.rs:142: Running "ip" "link" "set" "li_ca02f9_0001" "up"
[2018-06-19T11:46:53Z][DEBUG]src/bin/lithos_knot/setup_network.rs:155: Running "ip" "link" "set" "lo" "up"
[2018-06-19T11:46:53Z][DEBUG]src/bin/lithos_knot/setup_network.rs:168: Running "ip" "addr" "add" "10.0.0.1/24" "dev" "li-ca02f9-0001"
[2018-06-19T11:46:53Z][DEBUG]src/bin/lithos_knot/setup_network.rs:179: Running "ip" "link" "set" "li-ca02f9-0001" "up"
[2018-06-19T11:46:53Z][DEBUG]src/bin/lithos_knot/setup_network.rs:191: Running "ip" "route" "add" "default" "via" "10.0.0.1"
[2018-06-19T11:46:53Z][DEBUG]src/bin/lithos_knot/setup_network.rs:203: Running "arping" "-U" "10.0.0.1" "-c1"
[2018-06-19T11:46:53Z][ERROR] Fatal error running "container/container.0": arping failed: exited with code 1

```

If `lithos_tree` process is run with `strace` as follows, it gives us an indication why is this happening:


``` text
$ strace -f /usr/bin/lithos_tree
... unrelated output ommited ...

[pid 32223] execve("/usr/bin/arping", ["/usr/bin/arping", "-U", "10.0.0.1", "-c1"], 0x55d76cb2d0a0 /* 1 var */) = 0

... unrelated output ommited ...

[pid 32223] writev(2, [{iov_base="", iov_len=0}, {iov_base="arping: Too many args on command"..., iov_len=61}], 2arping: Too many args on command line. Expected at most one.
) = 61
[pid 32223] exit_group(1)               = ?
[pid 32223] +++ exited with 1 +++

```

The lithos configurations used are:

master config:

``` yaml
log_level: debug
devfs-dir: /dev
```

sandbox config:
``` yaml
allow-users: [0, 1, 100]
allow-groups: [0, 101]
image-dir: /var/lib/lithos/images
writable-paths:
  /var/log/container: /data
  /var/lib/container: /log
bridged-network:
  bridge: br0
  network: 10.0.0.0/24
  default_gateway: 10.0.0.1
```

process config:

``` yaml
container:
  image: container
  config: /config/container.yaml
  ip-addresses: [10.0.0.1]
```


It turns out that Alpine uses distribution of `arping` ([specifically this one by Thomas Habets](https://github.com/ThomasHabets/arping)) which is more strict with regards to argument order validation than `arping` distributions that are available in other (more mainstream) Linux distros (there it is usually provided by [iputils package](http://www.skbuff.net/iputils/iputils-current.tar.bz2)).

Simple reordering of arguments is sufficient to get it to work on both implementations of arping (I have tested only on Fedora and Alpine).

--- a/src/bin/lithos_knot/setup_network.rs
+++ b/src/bin/lithos_knot/setup_network.rs
@@ -198,8 +198,8 @@ fn _setup_bridged(sandbox: &SandboxConfig, _child: &ChildInstance, ip: IpAddr)
 
     let mut cmd = unshare::Command::new("/usr/bin/arping");
     cmd.arg("-U");
-    cmd.arg(&format!("{}", ip));
     cmd.arg("-c1");
+    cmd.arg(&format!("{}", ip));
     debug!("Running {}", cmd.display(&Style::short()));
     match cmd.status() {
         Ok(s) if s.success() => {}

